examples/demo/tutorial_walkthrough.md

```diff
@@ -0,0 +1,457 @@
+has [automatically been reloaded](https://github.com/fugue/fregot#watch):
+
+    Reloaded demo.rego
+
+(Note: If you didn't launch the REPL with `--watch`, you can manually reload all
+modified files with [`:reload`](https://github.com/fugue/fregot#reload).)
+
+You're still in error mode, so [quit](https://github.com/fugue/fregot#quit)
+error mode to return to normal REPL mode:
+
+    :quit
+
+### Enter Debug Mode Again
+
+We should make sure the rest of the policy is evaluated correctly. Activate the
+breakpoint again to return to debug mode:
+
+    deny
+
+You'll see the first query of `deny` again:
+
+    21|     ami = amis[ami]
+            ^^^^^^^^^^^^^^^
+
+Instead of stepping _into_ the next query, we'll step _over_ it with the
+[`:next`](https://github.com/fugue/fregot#next) command:
+
+    :next
+
+This brings us to line 22 of `demo.rego`, and you'll see this output, where the
+`invalid_ami` function determines whether the first AMI ID is whitelisted:
+
+    22|     invalid_ami(ami)
+            ^^^^^^^^^^^^^^^^
+
+Let's step over the next couple queries until we see the final results of the
+`deny` evaluation:
+
+    :next
+
+You'll see this output, as fregot evaluates the second AMI ID for validity:
+
+    22|     invalid_ami(ami)
+            ^^^^^^^^^^^^^^^^
+
+Step over the next query again:
+
+    :next
+
+You'll see this output:
+
+    (debug) = true
+    (debug) finished
+
+This means fregot has finished evaluating the input. `deny` returns true because
+`ami-atotallyfakeamiid` is not whitelisted -- exactly as expected!
+
+fregot has automatically exited debugging mode. Now, quit the REPL:
+
+    :quit
+
+Let's evaluate the policy as we did before -- this time, it should work
+properly:
+
+    fregot eval --input \
+      repl_demo_input.json \
+      'data.fregot.examples.demo.deny' demo.rego
+
+And it does!
+
+    [true]
+
+`deny` returns `true`, which means we've successfully tested the Terraform plan
+against the Rego policy and determined that the plan fails validation.
+
+But why stop there? Let's make some changes and use the
+[`:watch`](https://github.com/fugue/fregot#watch) command to automatically print
+the updated output.
+
+### Launch REPL Again
+
+Jump into the REPL again using the `--watch` flag:
+
+    fregot repl demo.rego --watch
+
+Load the policy:
+
+    :load demo.rego
+
+Set the input:
+
+    :input repl_demo_input.json
+
+### Set Watch on `deny`
+
+This time, use `:watch` to automatically re-evaluate `deny` any time we modify
+either the policy or input:
+
+    :watch deny
+
+We're going to modify the input to prove that `deny` works as expected. In
+`repl_demo_input.json`, replace `ami-atotallyfakeamiid` with
+`ami-04b9e92b5572fa0d1` on lines 39, 131, and 217 and save your changes. fregot
+automatically reloads the input and outputs the evaluation, so you'll see this:
+
+    Reloaded repl_demo_input.json 
+    {}
+
+The empty braces mean that `deny` does not return `true`. The Terraform plan
```

Comment:
I would write: ... means that `deny` returns no results -- concretely meaning that the request will not be denied.